test/core/kernel_gateway/get_updated_validators.js

```diff
@@ -0,0 +1,179 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const BN = require('bn.js');
+const web3 = require('../../test_lib/web3.js');
+
+const KernelGateway = artifacts.require('MockKernelGateway');
+const BlockStoreMock = artifacts.require('BlockStoreMock');
+
+contract('KernelGateway.getUpdatedValidators()', async (accounts) => {
+
+  let hash =
+    "0xb6a85955e3671040901a17db85b121550338ad1a0071ca13d196d19df31f56ca";
+
+  let randomHash =
+    "0x5fe50b260da6308036625b850b5d6ced6d0a9f814c0688bc91ffb7b7a3a54b67";
+
+  let activationHeight = new BN(1234);
+
+  let kernelGateway, auxiliaryBlockStore, initialKernelHash;
+
+  beforeEach(async function() {
+
+    initialKernelHash = web3.utils.sha3('kernelHash');
+    auxiliaryBlockStore = await BlockStoreMock.new();
+
+    kernelGateway = await KernelGateway.new(
+      accounts[1],
+      accounts[2],
+      auxiliaryBlockStore.address,
+      initialKernelHash,
+    );
+
+    await auxiliaryBlockStore.setKernelGateway(kernelGateway.address);
+    await kernelGateway.setOpenKernelHash(hash);
+    await kernelGateway.setOpenKernelActivationHeight(activationHeight);
+
+  });
+
+  it('should return blank array of validator addresses and weights when ' +
+    'kernel hash is not open kernel hash', async () => {
+
+    let result = await kernelGateway.getUpdatedValidators.call(randomHash);
+
+    assert.strictEqual(
+      result.updatedValidators_.length,
+      0,
+      "Contract must return blank array for validator addresses",
+    );
+
+    assert.strictEqual(
+      result.updatedWeights_.length,
+      0,
+      "Contract must return blank array for validator weights",
+    );
+
+  });
+
+  it('should return correct validator addresses and weights', async () => {
+
+    let validatorAddresses = [accounts[1], accounts[2]];
+    let validatorWeights = [new BN(100), new BN(150)];
+    let hash1 = web3.utils.sha3('hash1');
+
+    await kernelGateway.setKernel(
+      new BN(12),
+      randomHash,
+      validatorAddresses,
+      validatorWeights,
+      hash1,
+    );
+
+    await kernelGateway.setOpenKernelHash(hash1);
+    await auxiliaryBlockStore.activateKernel(hash1);
+
+    let result = await kernelGateway.getUpdatedValidators.call(hash1);
+
+    assert.strictEqual(
+      result.updatedValidators_.length,
+      2,
+      "Contract must return an array with 2 validator address",
+    );
+
+    assert.strictEqual(
+      result.updatedWeights_.length,
+      2,
+      "Contract must return an array with 2 validator weights",
+    );
+
+    for (let i = 0; i < result.length; i++) {
```

Comment:
This `for` loop can be replaced with `deepEquals`?

Reply:
I tried, `deepEqual` is not working.

Reply:
`deepEqual` works for the array, but in this case, it's array of 'bigNumber` which doesn't have `equal` unfortunately.  😞